packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -72,6 +73,21 @@ export type TreeObjectNode<
 	TypeName extends string = string,
 > = TreeNode & ObjectFromSchemaRecord<T> & WithType<TypeName>;
 
+/**
+ * Type utility for determining whether or not an implicit field schema has a default value.
+ *
+ * @privateRemarks
+ * TODO: Account for field schemas with default value providers.
+ * For now, this only captures field kinds that we know always have defaults - optional fields and identifier fields.
+ *
+ * @public
+ */
+export type HasDefault<T extends ImplicitFieldSchema> = T extends FieldSchema<
```

Comment:
I think this type has a bit too general name to be exported in our package scopes. `FieldHasDefault` seems like it would be better